desc/integrals/_bounce_utils.py

```diff
@@ -701,127 +850,323 @@ def interp_to_argmin_hard(h, points, knots, g, dg_dz, method="cubic"):
-        Matplotlib (fig, ax) tuple.
+    h : jnp.ndarray
+        Shape (..., num well).
 
     """
-    fig, ax = plt.subplots()
-    legend = {}
-    if include_knots:
-        for knot in ppoly.x:
-            _add2legend(
-                legend,
-                ax.axvline(
-                    x=knot, color="tab:blue", alpha=knot_transparency, label="knot"
-                ),
-            )
+    ext, g_ext = _get_extrema(knots, g, dg_dz, sentinel=0)
+    # Our softargmax(x) does the proper shift to compute softargmax(x - max(x)),
+    # but it's still not a good idea to compute over a large length scale, so we
+    # warn in docstring to choose upper sentinel properly.
+    argmin = softargmax(
+        beta * _where_for_fft_argmin(points, ext, g_ext, upper_sentinel),
+        axis=-1,
+    )
+    return jnp.linalg.vecdot(
+        argmin,  # shape is (..., num well, num extrema)
+        interp_rfft2(
+            T.eval1d(ext),
+            ext,
+            h[..., jnp.newaxis, :, :],
+            domain1=(0, 2 * jnp.pi / NFP),
+            axes=(-1, -2),
+        )[..., jnp.newaxis, :],
+        # adding axis to broadcast with num well axis
+    )
 
-    z = jnp.linspace(
-        start=setdefault(start, ppoly.x[0]),
-        stop=setdefault(stop, ppoly.x[-1]),
-        num=num,
+
+# TODO (#568): Generalize this beyond ζ = ϕ or just map to Clebsch with ϕ.
+def get_fieldline(alpha_0, iota, num_transit, period):
+    """Get sequence of poloidal coordinates A = (α₀, α₁, …, αₘ₋₁) of field line.
+
+    Parameters
+    ----------
+    alpha_0 : float
+        Starting field line poloidal label.
+    iota : jnp.ndarray
+        Shape (num rho, ).
+        Rotational transform normalized by 2π.
+    num_transit : float
+        Number of ``period``s to follow field line.
+    period : float
+        Toroidal period after which to update label.
+
+    Returns
+    -------
+    fieldline : jnp.ndarray
+        Shape (num rho, num transit) or broadcastable.
+        Sequence of poloidal coordinates A = (α₀, α₁, …, αₘ₋₁) that specify field line.
+
+    """
+    # Δϕ (∂α/∂ϕ) = Δϕ ι̅ = Δϕ ι/2π = Δϕ data["iota"]
+    return alpha_0 + period * (
+        iota if iota.size == 1 else iota[:, jnp.newaxis]
+    ) * jnp.arange(num_transit)
+
+
+def fourier_chebyshev(theta, iota, alpha, num_transit):
+    """Parameterize θ along field lines ``alpha``.
+
+    Parameters
+    ----------
+    theta : jnp.ndarray
+        Shape (num rho, X, Y) or (X, Y).
+        DESC coordinates θ sourced from the Clebsch coordinates
+        ``FourierChebyshevSeries.nodes(M,N,domain=(0,2*jnp.pi))``.
+        Use the ``Bounce2D.compute_theta`` method to obtain this.
+    iota : jnp.ndarray
+        Shape (num rho, ).
+        Rotational transform normalized by 2π.
+    alpha : float
+        Starting field line poloidal label.
+    num_transit : int
+        Number of toroidal transits to follow field line.
+
+    Returns
+    -------
+    T : PiecewiseChebyshevSeries
```

Comment:
Why do we need PiecewiseChebyshev? In what direction is it piecewise?

Reply:
[Once I have the mapping between coordinate systems](https://github.com/PlasmaControl/DESC/pull/1119#discussion_r1868374869), all I need to do is find the bounce points. This requires global nonlinear root finding. The best way to do this is to approximate the function in (alpha, zeta) coordinates and find the roots of this approximation.
We can either use cubic splines along the field line, or piecwise chebyshev series along the field line to do this

Reply:
So the piecewise-ness comes from changing alpha to alpha + 2piiota when you jump to a new field line?

Reply:
Yes